presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -266,26 +266,33 @@ private static PlanNodeStatsEstimate filterStatsForUnknownExpression(PlanNodeSta
                     .collect(ImmutableList.toImmutableList());
 
             if (!valuesEqualityStats.stream().allMatch(Optional::isPresent)) {
-                return visitExpression(node, context);
+                return Optional.empty();
             }
 
             PlanNodeStatsEstimate statsSum = valuesEqualityStats.stream()
                     .map(Optional::get)
                     .reduce(filterForFalseExpression().get(), PlanNodeStatsEstimateMath::addStatsAndSumDistinctValues);
 
             if (isNaN(statsSum.getOutputRowCount())) {
-                return visitExpression(node, context);
+                return Optional.empty();
             }
 
-            Symbol inValueSymbol = Symbol.from(node.getValue());
-            SymbolStatsEstimate symbolStats = input.getSymbolStatistics(inValueSymbol);
-            double notNullValuesBeforeIn = input.getOutputRowCount() * (1 - symbolStats.getNullsFraction());
+            Optional<Symbol> inValueSymbol = asSymbol(node.getValue());
+            SymbolStatsEstimate inValueStats = getExpressionStats(node.getValue());
+            if (Objects.equals(inValueStats, UNKNOWN_STATS)) {
```

Comment:
`UNKNOWN_STATS.equals(inValueStats)`

Reply:
Since this is a nit, feel free to leave it as is. But, please, just sate my curiosity. How's is the `Objects.equals(inValueStats, UNKNOWN_STATS)` style any better than a simple `equals`? Neither the constant, nor the `inValueStats` could be `null`. So the `Objects.equals` seems to be misleading. Since it implies that it's parameters may be `null`.

Reply:
I prefer `Objects.equals(a,b)` over `a.equals(b)` because the latter is asymmetric